conans/client/tools/env.py

```diff
@@ -111,3 +113,45 @@ def remove_from_path(command):
 
     with environment_append({"PATH": curpath}):
         yield
+
+
+def env_diff(cmd, only_diff):
+    known_path_lists = ("include", "lib", "libpath", "path")
+    if platform.system() == "Windows":
+        cmd += " && set"
+    else:
+        cmd += " && export"
+    ret = check_output_runner(cmd)
+    new_env = {}
+    for line in ret.splitlines():
+        line = line.strip()
+
+        if line == "\n" or not line:
+            continue
+        try:
+            name_var, value = line.split("=", 1)
+            name_var = str(name_var)
+            value = str(value)
+            new_value = value.split(os.pathsep) if name_var.lower() in known_path_lists else value
```

Comment:
Environment variables are case sensitive on non-windows platforms.

Reply:
great, I've changed the implementation to do upper-case only on Windows